_ml-commons-plugin/agents-tools/tools/query-planning-tool.md

```diff
@@ -0,0 +1,248 @@
+{: .warning}
+<!-- vale on -->
+
+The `QueryPlanningTool` generates an OpenSearch Query DSL query string from a natural language question.
+
+## Step 1: Enable the agentic search feature flag
+
+To use the `QueryPlanningTool`, you must first enable the `agentic_search_enabled` setting:
+
+```json
+PUT _cluster/settings
+{
+  "persistent" : {
+    "plugins.ml_commons.agentic_search_enabled" : true
+  }
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch responds with an acknowledgment:
+
+```json
+{
+  "acknowledged": true,
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "agentic_search_enabled": "true"
+      }
+    }
+  },
+  "transient": {}
+}
+```
+
+## Step 2: Create a connector for a model
+
+The following example request creates a connector for a model hosted on Amazon Bedrock:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "Amazon Bedrock Claude 3.7-sonnet connector",
+    "description": "connector for base agent with tools",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "parameters": {
+      "region": "us-west-2",
+      "service_name": "bedrock",
+      "model": "us.anthropic.claude-3-7-sonnet-20250219-v1:0",
+      "system_prompt":"please help answer the user question "
+    },
+    "credential": {
+      "access_key": "<YOUR_ACCESS_KEY>",
+      "secret_key": "<YOUR_SECRET_KEY>",
+      "session_token": "<YOUR_SESSION_TOKEN>"
+    },
+    "actions": [
+      {
+        "action_type": "predict",
+        "method": "POST",
+        "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/converse",
+        "headers": {
+          "content-type": "application/json"
+        },
+        "request_body": "{ \"system\": [{\"text\": \"${parameters.system_prompt}\"}], \"messages\": [${parameters._chat_history:-}{\"role\":\"user\",\"content\":[{\"text\":\"${parameters.user_prompt}\"}]}${parameters._interactions:-}]${parameters.tool_configs:-} }"
+      }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch responds with a connector ID:
+
+```json
+{
+  "connector_id": "NtjQi5gBOh0h20Y9GBWK"
+}
+```
+
+## Step 3: Register and deploy the model
+
+To register and deploy the model to OpenSearch, send the following request, providing the connector ID from the previous step:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "agentic search base model",
+    "function_name" : "remote",
+    "connector_id": "NtjQi5gBOh0h20Y9GBWK"
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch responds with a model ID:
+
+```json
+{
+  "task_id": "O9jQi5gBOh0h20Y9ghVC",
+  "status": "CREATED",
+  "model_id": "PNjQi5gBOh0h20Y9ghVY"
+}
+```
+
+## Step 4: Register a flow agent that will run the QueryPlanningTool
```

Comment:
We can use any agent, right? Should we not restrict it to flow then. We cam have the example using just flow agent

Reply:
added